<launch>
    <arg name="robot_name"/>
    <arg name="config_name"/>
    <arg name="rviz"                  default="true" />
    <arg name="performance_indices"   default="true" />
    <arg name="description_name"      default="ocs2_anymal_description"/>
    <arg name="target_command"        default="$(find ocs2_anymal_mpc)/config/$(arg config_name)/targetCommand.info"/>

    <group if="$(arg rviz)">
      <!-- Load robot description -->
      <include file="$(find ocs2_anymal_models)/launch/load_urdf.launch">
          <arg name="robot_name"         value="$(arg robot_name)"/>
          <arg name="description_name"   value="$(arg description_name)"/>
      </include>
      <!-- Launch visualization -->
      <include file="$(find ocs2_quadruped_interface)/launch/visualization.launch">
          <arg name="description_name"   value="$(arg description_name)"/>
      </include>
    </group>

    <node pkg="ocs2_anymal_loopshaping_mpc" type="ocs2_anymal_loopshaping_mpc_mpc_node" name="ocs2_anymal_loopshaping_mpc_mpc_node"
          output="screen" args="$(arg description_name) $(arg config_name)" launch-prefix=""/>

    <node pkg="ocs2_anymal_loopshaping_mpc" type="ocs2_anymal_loopshaping_mpc_dummy_mrt_node" name="ocs2_anymal_loopshaping_mpc_dummy_mrt_node"
          output="screen" args="$(arg description_name) $(arg config_name)" launch-prefix="gnome-terminal -- "/>

    <node pkg="ocs2_anymal_commands" type="gait_command_node" name="gait_command_node"
          output="screen" launch-prefix="gnome-terminal --"/>

    <node pkg="ocs2_anymal_commands" type="target_command_node" name="target_command_node"
        output="screen" args="$(arg target_command)" launch-prefix="gnome-terminal --"/>

    <!-- Launch Performance Indices Multi-plot -->
    <include if="$(arg performance_indices)" file="$(find ocs2_ros_interfaces)/launch/performance_indices.launch">
      <arg name="mpc_policy_topic_name"      value="anymal_mpc_policy"/>
    </include>
</launch>
